src/exercism/clojure_analyzer.clj

```diff
@@ -7,6 +7,28 @@
    [rewrite-clj.zip :as z])
   (:gen-class))
 
+;; Get clj-kondo static analysis:
+
+(defn ana [file]
+  (-> (clj-kondo/run! {:lint [file]
+                       :config {:output {:analysis true :format :edn}}}) :analysis))
+
+;; Check if user has defined the correct namespace and var:
+(defn ns-defined? [s file]
+  (some #(= (symbol s) %)
+        (map :name (:namespace-definitions (ana file)))))
+
+(defn var-defined? [s file]
+  (some #(= (symbol s) %)
+        (map :name (:var-definitions (ana file)))))
+ 
+(comment
+  (ns-defined? "two-fer" "resources/example.clj")
+  (var-defined? "two-fer" "resources/example.clj")
+  )
```

Comment:
What does this function call do?

Reply:
That's what is called a "Rich comment form". It's used for evaluating functions during interactive development, but the whole block is ignored by the compiler when the file is read.

Reply:
more info: https://github.com/PEZ/rich4clojure

Reply:
That's neat!

Reply:
I don't know how much Clojure you've done so I'm sorry if you know this already, but all development is done with an editor connected to the running program, which is built by evaluating functions (often one at a time) using editor hotkeys. So Clojure files are more than source code, they're like playgrounds.

Reply:
I've been doing Clojure in IntelliJ with the Cursive plugin, so I think that my workflow has been a bit different.